### PR TITLE
Test and lint workflows for backend

### PR DESCRIPTION
**Pull request by imilchev**
- Added test and lint workflows for backend
- Renamed lint workflow for frontend

### CODE REVIEW

**Comment by imilchev** [comment]
@kachar added workflows for linting and testing Go modules.

**Comment by imilchev** [comment]
I am having trouble running the workflow locally for some reason. Not sure if someone can confirm whether this will work

**Review by kachar — APPROVED** [review]
:rocket:

**Comment by kachar** [comment]
We'll see those in action in the followup PRs

**Comment by kachar** [comment]
```
run golangci-lint
  Running [/home/runner/golangci-lint-1.32.2-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=backend/graphql-gateway --timeout=3m0s] in [/home/runner/work/frontend/frontend/backend/graphql-gateway] ...
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: S1028: failed prerequisites: [(inspect@github.com/daritelska-platforma/graphql-gateway/pb/campaign, isgenerated@github.com/daritelska-platforma/graphql-gateway/pb/campaign): analysis skipped: errors in package: [/home/runner/work/frontend/frontend/backend/graphql-gateway/pb/campaign/campaign.pb.go:7:9: could not import common (pb/campaign/campaign.pb.go:7:2: package common is not in GOROOT (/opt/hostedtoolcache/go/1.16.4/x64/src/common))]]"
  level=warning msg="[runner] Can't run linter unused: buildir: failed to load package : could not load export data: no export data for \"common\""
  level=error msg="Running error: buildir: failed to load package : could not load export data: no export data for \"common\""
  
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 25938ms
```

https://github.com/daritelska-platforma/frontend/runs/2787516360?check_suite_focus=true

File: .github/workflows/backend-test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Setup golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```
Comment (kachar) [inline .github/workflows/backend-test.yml:18]: In the docker image we're using go version `1.15` if that matters

**Pull request merged**